ignite/metrics/ndcg.py

```diff
@@ -0,0 +1,81 @@
+from typing import Callable, Optional, Sequence, Union
+
+import torch
+
+from ignite.exceptions import NotComputableError
+from ignite.metrics.metric import Metric
+
+__all__ = ["NDCG"]
+
+
+class NDCG(Metric):
+    def __init__(
+        self,
+        output_transform: Callable = lambda x: x,
+        device: Union[str, torch.device] = torch.device("cpu"),
+        k: Optional[int] = None,
+        log_base: Union[int, float] = 2,
+    ):
+
+        self.log_base = log_base
+
+        self.k = k
+
+        super(NDCG, self).__init__(output_transform=output_transform, device=device)
+
+    def _dcg_sample_scores(
```

Comment:
Let's put dcg and `ndcg_sample_scores` functions out of NDCG class just define them before the class, such that they could be also used alone.